UnityProject/Assets/Scripts/Health/Living/LivingHealthBehaviour.cs

```diff
@@ -379,7 +381,23 @@ private void SyncFireStacks(float oldValue, float newValue)
 
 		if (attackType == AttackType.Fire)
 		{
-			SyncFireStacks(fireStacks, fireStacks+1);
+			// fire stacks should not exceed 20, and not apply if already at the cap
+			if (fireStacks <= maxFireStacks && !maxFireStacksReached)
```

Comment:
I think `maxFireStacksReached` bool isn't even needed here, a `if (fireStacks < maxFireStacks)` check should be enough

Reply:
You will forever set yourself on fire and generate new stacks without some flag to prevent yourself from also being a fire source. I puzzled with this one alot, The best long term solution might be to add a trait to fire to determine if its coming from the environment or the character. 

Though im still not sure of the best way to go about that.